src/bpfprogram.cpp

```diff
@@ -63,24 +63,29 @@ void BpfProgram::set_attach_target(const Probe &probe,
 
   const std::string &mod = probe.path;
   const std::string &fun = probe.attach_point;
+  const std::string &attach_target = !mod.empty() ? mod + ":" + fun : fun;
 
   const std::string &btf_fun = probe.type == ProbeType::iter ? "bpf_iter_" + fun
                                                              : fun;
   if (btf.get_btf_id(btf_fun, mod) < 0) {
-    std::string msg = "No BTF found for " + mod + ":" + fun;
+    const std::string &msg = "No BTF found for " + attach_target;
```

Comment:
qq: what good does it do making these temporary strings a ref type?

Reply:
None, I'm just sometimes too lazy to think about it and just use const ref everywhere.

Fixed, thanks.